### PR TITLE
Add CodePointTrie iai bench

### PR DESCRIPTION
**Pull request by zbraniecki**
This is a simple iai bench for CPT. Depends on #1851.

### CODE REVIEW

File: utils/codepointtrie/benches/iai_cpt.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    let sample = if mixed {
+        SAMPLE_STRING_MIXED
+    } else {
+        SAMPLE_STRING_LATIN1
+    };
+
+    let mut i: usize = 0;
+    for _ in 0..2439 {
+        for c in sample.chars() {
+            i += cpt.get(c as u32) as usize;
```
Comment (sffc) [inline utils/codepointtrie/benches/iai_cpt.rs:49]: Suggestion (optional): `i = i.wrapping_add(cpt.get(c as u32))`

**Review by sffc — DISMISSED** [review]

**Comment by sffc** [comment]
> First results:

Do you have an idea of how many of those cycles are "static" versus "dynamic"? Like, if you reduce the number of iterations in half, do the numbers also go down by about half?

**Comment by zbraniecki** [comment]
> Do you have an idea of how many of those cycles are "static" versus "dynamic"? Like, if you reduce the number of iterations in half, do the numbers also go down by about half?

With the updated code to remove unnecessary loops, around 60% of the instruction count is dynamic.

**Comment by zbraniecki** [comment]
Current results:

```
bench_iai_cpt_latin_fast
  Instructions:                 842 (No change)
  L1 Accesses:                  957 (No change)
  L2 Accesses:                    1 (No change)
  RAM Accesses:                  13 (No change)
  Estimated Cycles:            1417 (No change)

bench_iai_cpt_latin_small
  Instructions:                 841 (No change)
  L1 Accesses:                  954 (No change)
  L2 Accesses:                    3 (No change)
  RAM Accesses:                  13 (No change)
  Estimated Cycles:            1424 (No change)

bench_iai_cpt_mixed_fast
  Instructions:                1400 (No change)
  L1 Accesses:                 1586 (No change)
  L2 Accesses:                    3 (No change)
  RAM Accesses:                  30 (No change)
  Estimated Cycles:            2651 (No change)

bench_iai_cpt_mixed_small
  Instructions:                1477 (No change)
  L1 Accesses:                 1678 (No change)
  L2 Accesses:                    2 (No change)
  RAM Accesses:                  28 (No change)
  Estimated Cycles:            2668 (No change)
```

with `cpt.get` responsible for ~60% of the instructions.

**Review by Manishearth — DISMISSED** [review]
haven't had too close a look but i trust shane's review on this and a cursory glance seems good!

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**